python/ray/experimental/data/tests/test_dataset.py

```diff
@@ -385,18 +385,6 @@ def test_read_binary_files(ray_start_regular_shared):
             assert expected == item
         # Test metadata ops.
         assert ds.count() == 10
-        assert "bytes" in str(ds.schema()), ds
-        assert "bytes" in str(ds), ds
-
-
-def test_read_binary_files_with_paths(ray_start_regular_shared):
```

Comment:
Hmm I think we should definitely support paths still, since often metadata is encoded in file names.

Reply:
Really? That kind of breaks the data model here.

Reply:
I'll see if I can make it fit, maybe it will be fine.